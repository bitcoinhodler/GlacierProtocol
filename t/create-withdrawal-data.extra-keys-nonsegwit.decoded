{
  "txid": "a6e8fa4f654fc868f2e5e812a027065e7a4050bd92840a33a9398362d22d38e3",
  "hash": "a6e8fa4f654fc868f2e5e812a027065e7a4050bd92840a33a9398362d22d38e3",
  "version": 2,
  "size": 371,
  "vsize": 371,
  "weight": 1484,
  "locktime": 0,
  "vin": [
    {
      "txid": "9fa6ec376f68fe1df08c0ac7fa084eee1ce5236aa757f20effca12068544b262",
      "vout": 1,
      "scriptSig": {
        "asm": "0 304402205431f6426d7cf48678eed3e83ae956d199eab0c72b037a411cd230a3d23b023e02200ccbb6d1e31da29818ba7d43855e484629be26dc393275a43f079f85f5124da1[ALL] 3044022028a159daf40094968d8fa41bfd1bef43a9e3204c3a9fd0221504ea272215873a02207f8658bed5d5a86652e721df4923a7b53c2a223e4d7e456fb37d77e8218cc8c1[ALL] 522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae",
        "hex": "0047304402205431f6426d7cf48678eed3e83ae956d199eab0c72b037a411cd230a3d23b023e02200ccbb6d1e31da29818ba7d43855e484629be26dc393275a43f079f85f5124da101473044022028a159daf40094968d8fa41bfd1bef43a9e3204c3a9fd0221504ea272215873a02207f8658bed5d5a86652e721df4923a7b53c2a223e4d7e456fb37d77e8218cc8c1014c8b522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae"
      },
      "sequence": 4294967295
    }
  ],
  "vout": [
    {
      "value": 0.42496290,
      "n": 0,
      "scriptPubKey": {
        "asm": "OP_HASH160 8546e747c8094376b0a0af44bc5bbb8dd4f0218d OP_EQUAL",
        "desc": "addr(2N5PvoFj6adLPcqHAD24MxEZxvdQMGBLXow)#tjjjpqgh",
        "hex": "a9148546e747c8094376b0a0af44bc5bbb8dd4f0218d87",
        "address": "2N5PvoFj6adLPcqHAD24MxEZxvdQMGBLXow",
        "type": "scripthash"
      }
    }
  ]
}
