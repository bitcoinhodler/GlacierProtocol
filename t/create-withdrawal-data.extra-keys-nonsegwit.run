#!/bin/bash
set -e

# Test validation of extra keys. Only need 2, providing 4 as I would in the test withdrawal process


$GLACIERSCRIPT --regtest=$1 create-withdrawal-data << INPUT
y
y
y
y
y
y
2NAVkgszAvjZZwWpHpbCaxSaNqCCXPD8Bts
522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae
2N5PvoFj6adLPcqHAD24MxEZxvdQMGBLXow
1
0200000000010196a5d520248d33a0f3f979fbe49584b111ad93d24029d708aa97c6d0b0e9f6e401000000171600146b46cde42a4f55f0a79683394def0845075e6587ffffffff02107e88020000000017a9146a72e11d52ccec6f13f320af5a33004842fc570787a07f88020000000017a914bd3996bb9a6c401da7e65d4dbb3a9b978cd045ec870247304402200dc56503fbab60c8ab09389569d94380d213df13a1d3846c834d623e78a6b64802200ad7db18277b0217b734f6486c95b0298cca3091d089523e34115569654cd952012102a53529420ba4a5c2f4b86eedf213230e7e65ecb917a92cc37a36d51d847be25f00000000
4
cQCrT9Ncs9729ao7jbmAWrD9z7tF64s2yKzmD6nkiLAi9sXVZWAn
cP65UeSDZPiTLB6CBwasWv9oJYEjRgQXhswfwcT9HscEKDcEbgy4
cNYaH3onqrdMffpznhMMmrHn34fuTU59w5j8LM3H42VPcUsLeXy5
cRoydfinDRzzRQJp5niqJWukSYTfPJQM6ytqGN6nzonaz1mafgwD
10
y

y
INPUT
