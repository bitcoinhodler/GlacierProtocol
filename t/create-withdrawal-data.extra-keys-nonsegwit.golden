Are you running this on a computer WITHOUT a network connection of any kind? (y/n)?Have the wireless cards in this computer been physically removed? (y/n)?Are you running on battery power? (y/n)?Are you running on an operating system booted from a USB drive? (y/n)?Is your screen hidden from view of windows, cameras, and other people? (y/n)?Are smartphones and all other nearby devices turned off and in a Faraday bag? (y/n)?
You will need to enter several pieces of information to create a withdrawal transaction.


*** PLEASE BE SURE TO ENTER THE CORRECT DESTINATION ADDRESS ***


Source cold storage address: 
Redemption script for source cold storage address: 
Destination address: 
How many unspent transactions will you be using for this withdrawal? 
Please paste raw transaction #1 (hexadecimal format) with unspent outputs at the source address
OR
input a filename located in the current directory which contains the raw transaction data
(If the transaction data is over ~4000 characters long, you _must_ use a file.):

Transaction data found for source address.
TOTAL unspent amount for this raw transaction: 0.42500000 BTC

How many private keys will you be signing this transaction with (at least 2 required to complete transaction)?
#: Key #1: Key #2: Key #3: Key #4: 
Enter fee rate.
Satoshis per vbyte: 
Based on the provided rate, the fee will be 0.00004030 bitcoin.
Confirm? (y/n): 
Please enter the decimal amount (in bitcoin) to withdraw to the destination address.

Example: For 2.3 bitcoins, enter "2.3".

After a fee of 0.00004030, you have 0.42495970 bitcoins available to withdraw.

*** Technical note for experienced Bitcoin users:  If the withdrawal amount & fee are cumulatively less than the total amount of the unspent transactions, the remainder will be sent back to the same cold storage address as change. ***

Amount to send to 2N5PvoFj6adLPcqHAD24MxEZxvdQMGBLXow (leave blank to withdraw all funds stored in these unspent transactions): With no change output, the transaction fee is reduced, and 0.42496290 BTC will be sent to your destination.

Is this data correct?
*** WARNING: Incorrect data may lead to loss of funds ***

0.42500000 BTC in unspent inputs from cold storage address 2NAVkgszAvjZZwWpHpbCaxSaNqCCXPD8Bts
0.42496290 BTC going to destination address 2N5PvoFj6adLPcqHAD24MxEZxvdQMGBLXow
Fee amount: 0.00003710

Signing with private keys: 
cQCrT9Ncs9729ao7jbmAWrD9z7tF64s2yKzmD6nkiLAi9sXVZWAn
cP65UeSDZPiTLB6CBwasWv9oJYEjRgQXhswfwcT9HscEKDcEbgy4
cNYaH3onqrdMffpznhMMmrHn34fuTU59w5j8LM3H42VPcUsLeXy5
cRoydfinDRzzRQJp5niqJWukSYTfPJQM6ytqGN6nzonaz1mafgwD


Confirm? (y/n): 
Calculating transaction...

Final fee rate: 10.00 satoshis per vbyte

Raw signed transaction (hex):
020000000162b244850612caff0ef257a76a23e51cee4e08fac70a8cf01dfe686f37eca69f01000000fd1e010047304402205431f6426d7cf48678eed3e83ae956d199eab0c72b037a411cd230a3d23b023e02200ccbb6d1e31da29818ba7d43855e484629be26dc393275a43f079f85f5124da101473044022028a159daf40094968d8fa41bfd1bef43a9e3204c3a9fd0221504ea272215873a02207f8658bed5d5a86652e721df4923a7b53c2a223e4d7e456fb37d77e8218cc8c1014c8b522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354aeffffffff01227188020000000017a9148546e747c8094376b0a0af44bc5bbb8dd4f0218d8700000000

Transaction fingerprint (md5):
349e81dbdee6b2bcba5597d681c05ba9
QR code for transaction written to transaction.png
